torch/csrc/distributed/rpc/rpc_agent.cpp

```diff
@@ -78,6 +78,10 @@ std::shared_ptr<FutureMessage> RpcAgent::sendWithRetries(
 }
 
 void RpcAgent::retryExpiredRpcs() {
+  // storing futures and exception messages for non-retriable error-ed futures.
+  std::vector<std::pair<std::shared_ptr<FutureMessage>, std::string>>
```

Comment:
just checking: the idea is that we only want this cleared with the lock held? (per line 171)
(it's not clear that this would strictly be necessary?)

Does it make sense to bump the scope up to inside the while loop, so guaranteed not to leak these between runs?
(maybe it doesn't matter...)

Reply:
We should be able to clear the vector without the lock held now that its a local function var.